### PR TITLE
New package: sebescudie.GammaLauncher version 5.2.0-preview02

### PR DESCRIPTION
**Pull request by sebescudie**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/122471)

### CODE REVIEW

**Comment by sebescudie** [comment]
@microsoft-github-policy-service agree [company="sebescudie"]

**Comment by sebescudie** [comment]
@microsoft-github-policy-service agree company="sebescudie"

**Comment by stephengillie** [comment]
>  Manifest Error: Field usage requires verified publishers. [Icons]

File: manifests/s/sebescudie/GammaLauncher/5.2.0-preview02/sebescudie.GammaLauncher.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Tags:
+- vvvv gamma launcher
+ReleaseNotes: 5.2.0 preview 02
+ReleaseNotesUrl: https://github.com/sebescudie/GammaLauncher/releases/tag/5.2.0-preview02
+InstallationNotes: Have a delightful patching day
+Icons:
+- IconUrl: https://raw.githubusercontent.com/sebescudie/GammaLauncher/main/ico.png
+  IconFileType: png
+  IconResolution: custom
+  IconTheme: default
```
Comment (stephengillie) [inline manifests/s/sebescudie/GammaLauncher/5.2.0-preview02/sebescudie.GammaLauncher.locale.en-US.yaml:29]:
```suggestion
```
I don't know that we're accepting icons yet.

Comment (sebescudie) [reply manifests/s/sebescudie/GammaLauncher/5.2.0-preview02/sebescudie.GammaLauncher.locale.en-US.yaml:29]: Ok, thanks for the feedback.

**Comment by stephengillie** [comment]
Manual Validation ended with:
image

This package appears to have a dependency on `Microsoft.DotNet.DesktopRuntime.6`.

File: manifests/s/sebescudie/GammaLauncher/5.2.0-preview02/sebescudie.GammaLauncher.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Created using wingetcreate 1.5.3.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.5.0.schema.json
+
+PackageIdentifier: sebescudie.GammaLauncher
+PackageVersion: 5.2.0-preview02
+InstallerType: inno
```
Comment (stephengillie) [inline manifests/s/sebescudie/GammaLauncher/5.2.0-preview02/sebescudie.GammaLauncher.installer.yaml:6]:
```suggestion
InstallerType: inno
Dependencies:
  PackageDependencies:
    - PackageIdentifier: Microsoft.DotNet.DesktopRuntime.6
```

**Comment by stephengillie** [comment]
This package also appears to have a dependency on `cimgui`:
image

It seems that `cimgui` hasn't been added to the WInGet repo yet, as I can't find it when searching.

File: manifests/s/sebescudie/GammaLauncher/5.2.0-preview02/sebescudie.GammaLauncher.installer.yaml (hunk 2 above)
Comment (sebescudie) [reply manifests/s/sebescudie/GammaLauncher/5.2.0-preview02/sebescudie.GammaLauncher.installer.yaml:6]: Thanks for the pointer, will commit that one once the `cimgui` issue is solved.

**Comment by sebescudie** [comment]
> This package also appears to have a dependency on `cimgui`: image
> 
> It seems that `cimgui` hasn't been added to the WInGet repo yet, as I can't find it when searching.

That's very odd, `cimgui.dll` is bundled with the program, so this looks like one of its dependencies is missing. Have never encountered this issue though, even after trying to install locally from the manifest file.

Could you share some information about the environment in which you're running this test in?

**Comment by sebescudie** [comment]
Just tried running `winget install --manifest [...]` on a fresh W11 VM. It looks like the changes you suggested in `sebescudie.GammaLauncher.installer` does not install the .NET6 dependency. The installation finishes properly, but when trying to open the app it prompts me that .NET6 is missing, as you reported in https://github.com/microsoft/winget-pkgs/pull/122471#issuecomment-1760391148.

I manually installed .NET6 and after that the app ran without issue.

Let me know how I can help troubleshoot this!

Thanks.

**Comment by stephengillie** [comment]
That's strange - a script adds the dependency to the manifest and validates locally. Then I copy that snippet into the PR. 
- Could you upload the manifest files used?  
- What is the file name or URL for the Dotnet package that was downloaded? 
- There are several .NET runtimes and frameworks, It's important to find the correct PackageIdentifier, to specify as a dependency. 

CimGUI is the same situation, but I couldn't find them in our repo. 
- Their repo appears to be here: https://github.com/cimgui/cimgui
- A manifest will have to be made and submitted for this package as well. 
- Then it can be added as a dependency here.

**Comment by sebescudie** [comment]
Hey, thanks for the heads up.

> * Could you upload the manifest files used?

Sure, they're all in this zip folder.
[gamma_launcher_manifest.zip](https://github.com/microsoft/winget-pkgs/files/12905346/gamma_launcher_manifest.zip)


> * What is the file name or URL for the Dotnet package that was downloaded?

`windowsdesktop-runtime-6.0.23-win-x64.exe`, link [here](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-6.0.23-windows-x64-installer?cid=getdotnetcore)



> * There are several .NET runtimes and frameworks, It's important to find the correct PackageIdentifier, to specify as a dependenc

The package you provided in your commit suggestion seems to be what I'm after, as per [wininstall.app](https://winget.run/pkg/Microsoft/DotNet.DesktopRuntime.6). It's just that the `winget` command does not take care of installing it when I try to run from manifest with your commit suggestion.

> CimGUI is the same situation, but I couldn't find them in our repo.

It's odd that you need an extra dependency for this one as everything is supposed to come bundled with the GammaLauncher app. If you give more information about your testing environment I could try to repro that and see if DependencyWalker gives more clues, because that's the first time I've heard of such an error.

Thanks for your patience and for looking into it!

**Comment by stephengillie** [comment]
> The package you provided in your commit suggestion seems to be what I'm after, as per [wininstall.app](https://winget.run/pkg/Microsoft/DotNet.DesktopRuntime.6). It's just that the `winget` command does not take care of installing it when I try to run from manifest with your commit suggestion.

Which version of WinGet are you running? It sounds like you might have a version that doesn't like dependencies. 

> It's odd that you need an extra dependency for this one as everything is supposed to come bundled with the GammaLauncher app. If you give more information about your testing environment I could try to repro that and see if DependencyWalker gives more clues, because that's the first time I've heard of such an error.

It's a fresh install of Windows 10, running inside a VM. More info [here](https://github.com/microsoft/winget-pkgs/issues/107639), including a `winget list` output.  

> Thanks for your patience and for looking into it!

Happy to help.

**Comment by sebescudie** [comment]
> Which version of WinGet are you running? It sounds like you might have a version that doesn't like dependencies.

Thanks for the pointer. Ran Windows Update on my W11 VM (now 22H2 22621.2428) and it indeed installs the NET6 dependency. The app then starts without issue.

> It's a fresh install of Windows 10, running inside a VM. More info [here](https://github.com/microsoft/winget-pkgs/issues/107639), including a `winget list` output.

Noted, I will try to spin up a similar VM and check this `cimgui` issue. Will report back.

**Comment by sebescudie** [comment]
Hey there,

Just ran the installer from the manifest on fresh W10 VM with latest Windows Update applied, the app installs and launches without trouble.

I just installed `winget` there using [this link](https://apps.microsoft.com/detail/9NBLGGH4NNS1) because it was not available out of the box on W10.

If you can repro the error on your end, could you please share the full stack trace that comes with Exception window you posted in https://github.com/microsoft/winget-pkgs/pull/122471#issuecomment-1761198829?

Maybe Windows' Event Log also has interesting information.

Thanks in advance!

**Comment by stephengillie** [comment]
Here's the stack trace: 
[sebescudie.GammaLauncher.txt](https://github.com/microsoft/winget-pkgs/files/13036112/sebescudie.GammaLauncher.txt)

**Comment by sebescudie** [comment]
Thanks, have [reported the error](https://discourse.vvvv.org/t/validation-failing-for-gammalauncher-winget-package/21955), will report back if can find a fix.

**Comment by sebescudie** [comment]
One question : could you check that `cimgui.dll` is indeed present next to `GammaLauncher.exe` in `C:\Program Files\GammaLauncher`? Maybe Windows Defender ate it for some reason. 

Thanks in advance!

**Comment by stephengillie** [comment]
Yes, it's there. Even double-clicking on the program, right next to the DLL, it still errors:
image

**Comment by sebescudie** [comment]
Thanks for the feedback. My next guess would be a missing `vcredist`. Is there any way you could show the list of `vcredist` you have installed on your test machine? I have those here. I could try to get the same as you on my VM, find which one could be missing and add it as a dependency in the installer.

```
Name                                                               Version
----                                                               -------
Microsoft Visual C++ 2010  x64 Redistributable - 10.0.40219        10.0.40219
Microsoft Visual C++ 2008 Redistributable - x64 9.0.30729.6161     9.0.30729.6161
Microsoft Visual C++ 2008 Redistributable - x64 9.0.30729.17       9.0.30729
Microsoft Visual C++ 2005 Redistributable (x64)                    8.0.61000
Microsoft Visual C++ 2013 Redistributable (x64) - 12.0.40664       12.0.40664.0
Microsoft Visual C++ 2013 Redistributable (x64) - 12.0.30501       12.0.30501.0
Microsoft Visual C++ 2012 Redistributable (x86) - 11.0.61030       11.0.61030.0
Microsoft Visual C++ 2015-2022 Redistributable (x86) - 14.34.31938 14.34.31938.0
Microsoft Visual C++ 2005 Redistributable                          8.0.61001
Microsoft Visual C++ 2008 Redistributable - x86 9.0.30729.6161     9.0.30729.6161
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.40664       12.0.40664.0
Microsoft Visual C++ 2012 Redistributable (x64) - 11.0.61030       11.0.61030.0
Microsoft Visual C++ 2015-2022 Redistributable (x64) - 14.34.31938 14.34.31938.0
Microsoft Visual C++ 2010  x86 Redistributable - 10.0.40219        10.0.40219
Microsoft Visual C++ 2013 Redistributable (x86) - 12.0.30501       12.0.30501.0
```

Thanks in advance!

**Comment by stephengillie** [comment]
> Thanks for the feedback. My next guess would be a missing `vcredist`. Is there any way you could show the list of `vcredist` you have installed on your test machine? 

> More info [here](https://github.com/microsoft/winget-pkgs/issues/107639), including a `winget list` output.

**Comment by sebescudie** [comment]
Was able to repro locally by uninstalling all `vcredist`. Adding a dependency to `Microsoft.VCRedist.2015+.x64` in the installer file solves the issue here.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**